### DISCUSSION TITLE
Introduce a way to Automatically make search params keys kebab case

### DISCUSSION THREAD

**Discussion by neefrehman** [47ng/nuqs#688] [Ideas] [answered]
kebab-case is typically the convention to follow with URL strings.  Despite this, it's more ergonomic to work with camelCase strings in code so that can map to JS symbols more easily. An example of this could be:

```ts
const searchParams = {
  "comments-after": parseAsString,
  "comment-count": parseAsInteger
}

const serializer = createSerializer(searchParams)

// ...

const [
  { "comments-after": commentsAfter, "comment-count": commentCount },
  setState
] = useQueryStates(searchParams)

const href = serializer({ "comments-after": cursor, "comment-count": 10 })
```

The DX of which I think would be improved if we were to just write:

```ts
const searchParams = {
  commentsAfter: parseAsString,
  commentCount: parseAsInteger
}

const serializer = createSerializer(searchParams)

// ...

const [{ commentsAfter, commentCount }, setState] = useQueryStates(searchParams)

const href = serializer({ commentsAfter: cursor, commentCount: 10 })
```

Though, with the latter option we now use camel case strings in our URL, breaking convention. It would be nice if `nuqs` offered a way[^1] to automatically kebab-case our keys, so that we can have both good DX, and conventional URL strings. Curious to hear if you agree!

[^1]: I appreciate that currently we could achieve this with `urlKeys`, but that currently isn't supported by serializers, and still I think that it would be nice to avoid any manual intervention to meet convention.

**Comment by franky47** [comment, marked as answer]
I believe this should be the job of the `urlKey`, which could use a little more love. I shipped it quickly as I needed it for a client project, but eventually this should be a first-class API.

One drawback of kebab-case is the extra dash: if URL space is precious, any character counts.

I like your proposal in #689, let's move the discussion there.

Related discussion: #672
